### PR TITLE
Add `unknown_t` type for unknown types

### LINKED ISSUE #1497: project saved in 0.2.1, opened with 0.3.0-git, base types cannot be found
Still for me to investigate more.

Quickly, what i did was creating a project in 0.2.1, then load with current dev and saving the project again. When saving it was saying that some of the base types could not be found. I saved anyway. When re-opening the project, there were multiple errors of parsing like:

```
error parsing "mbsinit" func arg type "(null)*": Wrongly formed ............
Cannot parse type_descriptor's type field
```

### PR DESCRIPTION
**Pull request by XVilka**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

During the import of the old projects some types were formed incorrectly due to how types were working before. In case of the missing type e.g. for function argument or structure or a typedef we now assign it to the `unknown_t` type, which is integer of platform current bitness.

**Test plan**

CI is green

**Closing issues**

Should fix https://github.com/rizinorg/rizin/issues/1497

### CODE REVIEW

**Review by ret2libc — APPROVED** [review]

**Comment by wargio** [comment]
LGTM

File: librz/type/type.c [hunk 1]
```diff
@@ -900,35 +901,35 @@ static char *type_as_string_identifier_decl(const RzTypeDB *typedb, RZ_NONNULL c
...
+			// if not - just the name
+			if (!strncmp(type->identifier.name, "anonymous ", 10)) {
+				char *btypestr = btype->kind == RZ_BASE_TYPE_KIND_TYPEDEF ? strdup(btype->name) : rz_type_db_base_type_as_string(typedb, btype);
+				if (type->identifier.is_const) {
+					rz_strbuf_appendf(buf, "const %s%s", btypestr, separator);
+				} else {
+					rz_strbuf_append(buf, btypestr);
+					rz_strbuf_append(buf, separator);
+				}
+				free(btypestr);
```
Comment (ret2libc) [inline librz/type/type.c:916]: this seems like a case worth a new unit test, if there is not one already

**Review by ret2libc — CHANGES_REQUESTED** [review]

File: librz/type/type.c (hunk 1 above)
Comment (XVilka) [reply librz/type/type.c:916]: Which one? The `unknown_t` substitution for the unknown types?

Comment (ret2libc) [reply librz/type/type.c:916]: Ignore me....

**Review by ret2libc — APPROVED** [review]

**Review by ret2libc — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Comment by ret2libc** [comment]
Tested this and it seems to fix the specific problem in #1497 but https://github.com/rizinorg/rizin/issues/1560 still happens and actually with this PR it just says a lot of `Cannot find base type "unknown_t"`. JFYI.

**Pull request merged**